cdap-explore-client/src/main/java/co/cask/cdap/explore/client/ExploreHttpClient.java

```diff
@@ -110,6 +110,18 @@ protected String getUserId() {
     return null;
   }
 
+  @Nullable
+  protected String getUserPrincipal() {
```

Comment:
I will add it to https://github.com/caskdata/cdap/pull/7972/files#diff-818f9f36bd99e1e10bf573e17c876d68R120